runtime/interpreter/value.go

```diff
@@ -1348,6 +1348,33 @@ func (v *ArrayValue) RemoveLast(interpreter *Interpreter, getLocationRange func(
 	return v.Remove(interpreter, getLocationRange, v.Count()-1)
 }
 
+func (v *ArrayValue) IndexOf(interpreter *Interpreter, getLocationRange func() LocationRange, needleValue Value) OptionalValue {
+
+	needleEquatable, ok := needleValue.(EquatableValue)
+	if !ok {
+		panic(errors.NewUnreachableError())
+	}
```

Comment:
At run-time the value is required to be equatable, so this must also be checked statically. See the comment for the implementation in `sema`

Reply:
I copied this code block from Contains, since it was very similiar and it is not implemented there. Is that wrong then?

Reply:
Right, I missed that 👍